src/map/latent_effect_container.cpp

```diff
@@ -1674,7 +1675,23 @@ void CLatentEffectContainer::CheckLatentsZone()
             }
             break;
         case LATENT_WEATHER_ELEMENT:
-            if (zoneutils::GetWeatherElement(zoneutils::GetZone(m_POwner->getZone())->GetWeather()) == m_LatentEffectList.at(i)->GetConditionsValue())
+            // In the default case we need to pull weather from zone/player
+            if (weather == MAX_WEATHER_ID)
```

Comment:
In attempts to avoid using an on-attainable value we could just do:
WEATHER GetWeather(CBattleEntity* PEntity, bool ignoreScholar)
WEATHER GetWeather(CBattleEntity* PEntity, bool ignoreScholar, uint16 zoneWeather)

Where the first method pulls the weather via zoneutils::GetZone(PEntity->getZone())->GetWeather() and calls the second